modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -674,31 +598,18 @@ protected ClassLoader getContextClassLoader() {
     return Thread.currentThread().getContextClassLoader();
   }
 
-  protected void parseFromTextExpression(ParameterModel parameter, DslElementSyntax dsl, Supplier<TypeConverter> typeConverter) {
-    parseFromTextExpression(parameter.getName(), dsl, typeConverter);
-  }
-
-  protected void parseFromTextExpression(String key, DslElementSyntax paramDsl, Supplier<TypeConverter> typeConverter) {
-    addParameter(getChildKey(key), fromChildConfiguration(String.class).withWrapperIdentifier(paramDsl.getElementName()));
-
-    addDefinition(definitionBuilder
-        .withIdentifier(paramDsl.getElementName())
-        .withTypeDefinition(fromType(String.class))
-        .withTypeConverter(typeConverter.get())
-        .build());
-  }
-
   /**
    * Registers a definition for parsing the given {@code parameterModel} as an element attribute
    *
    * @param parameterModel a {@link ParameterModel}
    * @return an {@link AttributeDefinition.Builder}
    */
-  protected AttributeDefinition.Builder parseAttributeParameter(ParameterModel parameterModel) {
+  protected AttributeDefinition.Builder parseSimpleParameter(ParameterModel parameterModel) {
```

Comment:
just because the parameter is being parsed from an attribute, it doesn't mean it's simple (or at least what the SDK referts to as a simple parameter)